discovery_engine/lib/src/domain/models/active_search.dart

```diff
@@ -12,28 +12,46 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:equatable/equatable.dart';
 import 'package:hive/hive.dart'
     show HiveType, HiveField, TypeAdapter, BinaryReader, BinaryWriter;
+import 'package:xayn_discovery_engine/src/api/models/active_search.dart' as api;
 import 'package:xayn_discovery_engine/src/domain/repository/type_id.dart'
     show searchTypeId, searchByTypeId;
 
-part 'active_search.freezed.dart';
 part 'active_search.g.dart';
 
 /// [ActiveSearch] represents attributes of a performed search.
-@freezed
-class ActiveSearch with _$ActiveSearch {
-  @HiveType(typeId: searchTypeId)
-  const factory ActiveSearch({
-    @HiveField(0) required String searchTerm,
-    @HiveField(1) required int requestedPageNb,
-    @HiveField(2) required int pageSize,
-    @HiveField(3, defaultValue: SearchBy.query) required SearchBy searchBy,
-  }) = _ActiveSearch;
-
-  factory ActiveSearch.fromJson(Map<String, Object?> json) =>
-      _$ActiveSearchFromJson(json);
+@HiveType(typeId: searchTypeId)
+class ActiveSearch with EquatableMixin {
+  @HiveField(0)
+  final String searchTerm;
+  @HiveField(1)
+  final int requestedPageNb;
+  @HiveField(2)
+  final int pageSize;
+  @HiveField(3, defaultValue: SearchBy.query)
+  final SearchBy searchBy;
+
+  const ActiveSearch({
+    required this.searchTerm,
+    required this.requestedPageNb,
+    required this.pageSize,
+    required this.searchBy,
+  });
+
+  ActiveSearch nextPageSearch() => ActiveSearch(
+        searchTerm: searchTerm,
+        requestedPageNb: requestedPageNb + 1,
+        pageSize: pageSize,
+        searchBy: searchBy,
+      );
+
+  api.ActiveSearch toApiRepr() =>
```

Comment:
This object should be completely agnostic of the api models we are currently using

Reply:
It's the same approach as currently used by `Document`, it that wouldn't already have been this way in `Document` I would have placed it into `api/models/active_search.dart`. 

When doing so should I also move to `toApiDocument` method on `Document` from `domain/models` to `api/models`?

Reply:
Through it also add some complexity  as you have to either write `ActiveSearch.fromInternalRepr(search)` instead of `search.toApiRepr()` or have to write a extension trait you then also have to import.

Reply:
It's now a extension method.